src/main/java/org/commcare/cases/instance/IndexedFixtureChildElement.java

```diff
@@ -0,0 +1,74 @@
+package org.commcare.cases.instance;
+
+import org.commcare.cases.model.StorageIndexedTreeElementModel;
+import org.javarosa.core.model.instance.TreeElement;
+import org.javarosa.core.model.instance.TreeReference;
+
+/**
+ * Child TreeElement of an indexed fixture whose data is loaded from a DB.
+ *
+ * i.e. 'product' of "instance('product-list')/products/product"
+ *
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+public class IndexedFixtureChildElement extends StorageBackedChildElement<StorageIndexedTreeElementModel> {
+    private TreeElement empty;
+
+    protected IndexedFixtureChildElement(StorageInstanceTreeElement<StorageIndexedTreeElementModel, ?> parent,
+                                         int mult, int recordId) {
+        super(parent, mult, recordId, parent.getName(), parent.getChildHintName());
+    }
+
+    /**
+     * Template constructor (For elements that need to create reference nodesets but never look up values)
+     */
+    private IndexedFixtureChildElement(StorageInstanceTreeElement<StorageIndexedTreeElementModel, ?> parent) {
```

Comment:
Since this is only ever called by `buildFixtureChildTemplate` I would be in favor of not making it a constructor and just putting this code into that method.